clarity_ext/service/file_service.py

```diff
@@ -266,6 +266,15 @@ def search_or_create(self, file_handle, mode='ab', extension="",
         return self._local_shared_file(artifact, filename, mode=mode, extension=extension,
                                        modify_attached=modify_attached)
 
+    def check_file_extension(self, file_handle, required_extension=None, filename_contains=None):
+        artifact = self._artifact_by_name(file_handle, filename=filename_contains)
+        file_name = os.path.basename(artifact.file_name)
+        _, actual_extension = os.path.splitext(file_name)
+        if not required_extension in file_name:
```

Comment:
Wouldn't it be better to check the extension, to avoid false positives? Since you split it out anyway?